Application/EdFi.Ods.Extensions.TPDM/Artifacts/MsSql/Data/Security/1010-TPDM-ResourceClaims.sql

```diff
@@ -1,465 +1,6178 @@
-
-SELECT @AuthorizationStrategyId  = (SELECT AuthorizationStrategyId FROM [dbo].[AuthorizationStrategies] WHERE AuthorizationStrategyName = 'NoFurtherAuthorizationRequired');
-
---- These Resources need explicit metadata ---
-INSERT INTO [dbo].[ResourceClaimAuthorizationMetadatas]
-    ([Action_ActionId]
-    ,[AuthorizationStrategy_AuthorizationStrategyId]
-    ,[ResourceClaim_ResourceClaimId]
-    ,[ValidationRuleSetName])
-SELECT ac.ActionId, @AuthorizationStrategyId, ResourceClaimId, null
-FROM [dbo].[ResourceClaims]
-CROSS APPLY
-    (SELECT ActionId
-    FROM [dbo].[Actions]
-    WHERE ActionName IN ('Create', 'Read', 'Update', 'Delete')) AS ac
-WHERE ResourceName IN ('anonymizedStudent'
-, 'anonymizedStudentAssessments'
-, 'performanceMeasure'
-, 'performanceMeasureCourseAssociation'
-, 'professionalDevelopmentEvent'
-, 'recruitmentEvent'
-, 'staffTeacherPreparationProviderAssociation'
-, 'anonymizedStudentAssessment'
-, 'talentManagementGoal'
-, 'teacherCandidate'
-, 'teacherCandidateStudentGrowthMeasure'
-, 'teacherCandidateTeacherPreparationProviderAssociation'
-, 'anonymizedStudentAcademicRecord'
-, 'anonymizedStudentAssessmentCourseAssociation'
-, 'anonymizedStudentAssessmentSectionAssociation'
-, 'anonymizedStudentCourseAssociation'
-, 'anonymizedStudentCourseTranscript'
-, 'anonymizedStudentEducationOrganizationAssociation'
-, 'anonymizedStudentSectionAssociation'
-, 'applicant'
-, 'applicantProspectAssociation'
-, 'application'
-, 'applicationEvent'
-, 'completerAsStaffAssociation'
-, 'employmentEvent'
-, 'employmentSeparationEvent'
-, 'openStaffPositionEvent'
-, 'prospect'
-, 'staffApplicantAssociation'
-, 'fieldworkExperienceSectionAssociation'
-, 'staffProspectAssociation'
-, 'staffStudentGrowthMeasure'
-, 'staffStudentGrowthMeasureCourseAssociation'
-, 'staffStudentGrowthMeasureEducationOrganizationAssociation'
-, 'staffStudentGrowthMeasureSectionAssociation'
-, 'staffTeacherPreparationProviderProgramAssociation'
-, 'teacherCandidateTeacherPreparationProviderProgramAssociation'
-, 'teacherCandidateAcademicRecord'
-, 'teacherCandidateCourseTranscript'
-, 'teacherCandidateStaffAssociation'
-, 'teacherCandidateStudentGrowthMeasureCourseAssociation'
-, 'teacherCandidateStudentGrowthMeasureEducationOrganizationAssociation'
-, 'teacherCandidateStudentGrowthMeasureSectionAssociation'
-, 'teacherPreparationProviderProgram'
-, 'professionalDevelopmentEventAttendance'
-);
+BEGIN
+    DECLARE 
+        @applicationId AS INT,
+        @claimId AS INT,
+        @claimName AS nvarchar(max),
+        @parentResourceClaimId AS INT,
+        @existingParentResourceClaimId AS INT,
+        @claimSetId AS INT,
+        @claimSetName AS nvarchar(max),
+        @authorizationStrategyId AS INT,
+        @msg AS nvarchar(max),
+        -- Well-known values
+        @createActionId AS INT = 1,
```

Comment:
cant make this assumption due to deployments can be different. need to set these values using a lookup.

Reply:
I think that with these values in particular that people would have no reason to introduce additional actions or to reassign their values. However, this is a database table out in the wild with an identity column, so anything is possible. I will make the change to the scripts to look up these action Ids.